ouroboros-consensus/src/Ouroboros/Consensus/HardFork/Combinator/Degenerate.hs

```diff
@@ -304,20 +319,23 @@ instance NoHardForks b => BasicEnvelopeValidation (DegenFork b) where
 instance NoHardForks b => ValidateEnvelope (DegenFork b) where
   type OtherHeaderEnvelopeError (DegenFork b) = OtherHeaderEnvelopeError (HardForkBlock '[b])
 
+{-
```

Comment:
Comment back in

Reply:
Turns out this was in fact wrong. It was delegating to `b`, not to the HFC. It's now

```haskell
  additionalEnvelopeChecks cfg view (DHdr hdr) =
      additionalEnvelopeChecks (castTopLevelConfig cfg) view hdr
```